src/DynamicNLPModels.jl

```diff
@@ -1369,11 +1408,11 @@ function _build_H_blocks(Q, R, block_A::M, block_B::M, S, Qf, K, s0, N) where {T
 
     LinearAlgebra.axpy!(1.0, Q, quad_term)
     LinearAlgebra.axpy!(1.0, SK, quad_term)
-    LinearAlgebra.axpy!(1.0, SK', quad_term)
+    quad_term .+= SK'
```

Comment:
Can we add a comment here to explain why we are using the broadcast operator for the transpose operation?